vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion.cc

```diff
@@ -251,14 +243,34 @@ void Promotion::OnGetAllPromotions(
       continue;
     }
 
-    promotions.insert(std::make_pair(item->id, item->Clone()));
+    promotions_ui.push_back(item->Clone());
 
     ledger_->database()->SavePromotion(
         item->Clone(),
         [](const ledger::Result _){});
   }
 
-  ledger::PromotionList promotions_ui = GetPromotionsForUI(promotions);
+  bool found;
+  for (const auto& promotion : promotions) {
```

Comment:
I think we should probably add a comment above this loop describing what's going on, since it's not immediately obvious from the code.

Reply:
added